raven/versioning.py

```diff
@@ -0,0 +1,33 @@
+from __future__ import absolute_import
+
+__all__ = ('fetch_git_sha', 'fetch_package_version')
+
+import os.path
+import pkg_resources
+
+from .exceptions import InvalidGitRepository
+
+
+def fetch_git_sha(path, head='master'):
+    """
+    >>> fetch_git_sha(os.path.dirname(__file__))
+    """
+    revision_file = os.path.join(path, '.git', 'refs', 'heads', 'master')
```

Comment:
I'm not a git expert, but I recall having issues with using .git/HEAD

Definitely would love input here, as I honestly have no desire to research the many ways this can fail.

(I fixed the ref to s/master/HEAD/)